src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -1288,50 +768,1631 @@ TEST_F(TrajectoryPlannerTest, velocity_trajectory_curved_path_low_acceleration)
+
+    trajectory.trajectory_elements[0].angular_speed = 0;
+    trajectory.trajectory_elements[1].angular_speed = 3;
+    trajectory.trajectory_elements[2].angular_speed = 2;
+    trajectory.trajectory_elements[3].angular_speed = 0;
+
+    TrajectoryPlannerGenerationStatus_t status =
+        app_trajectory_planner_modifyTrajectoryToBackwardsContinuous(&trajectory,
+                                                                     segment_lengths);
+    EXPECT_EQ(status, OK);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[0].angular_speed, 0, 0.0001);
+    EXPECT_NEAR(trajectory.trajectory_elements[1].angular_speed, sqrt(4), 0.0001);
+    EXPECT_NEAR(trajectory.trajectory_elements[2].angular_speed, sqrt(2), 0.0001);
+    EXPECT_NEAR(trajectory.trajectory_elements[3].angular_speed, 0, 0.0001);
+}
+
+TEST_F(TrajectoryPlannerTest,
+       test_backwards_continuity_variable_segment_length_continuous_angular)
+{
+    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+    const unsigned int num_segments = 4;
+
+
+    TrajectorySegment_t segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    segment_lengths[0].angular_segment_length = 1.0;
+    segment_lengths[1].angular_segment_length = 2.0;
+    segment_lengths[2].angular_segment_length = 3.0;
+
+
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.max_allowable_angular_acceleration = 1;
+    path_parameters.max_allowable_angular_speed        = 1;
+    path_parameters.max_allowable_linear_acceleration  = 1;
+    path_parameters.max_allowable_linear_speed         = 1;
+    path_parameters.num_segments                       = num_segments;
+    path_parameters.initial_linear_speed               = sqrt(7);
+    path_parameters.final_linear_speed                 = 1;
+
+    float speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+    PositionTrajectory_t trajectory;
+    PositionTrajectoryElement_t trajectory_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    trajectory.path_parameters     = path_parameters;
+    trajectory.trajectory_elements = trajectory_elements;
+
+    trajectory.trajectory_elements[0].angular_speed = 0;
+    trajectory.trajectory_elements[1].angular_speed = 3;
+    trajectory.trajectory_elements[2].angular_speed = 10;
+    trajectory.trajectory_elements[3].angular_speed = 0;
+
+    TrajectoryPlannerGenerationStatus_t status =
+        app_trajectory_planner_modifyTrajectoryToBackwardsContinuous(&trajectory,
+                                                                     segment_lengths);
+    EXPECT_EQ(status, OK);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[0].angular_speed, 0, 0.0001);
+    EXPECT_NEAR(trajectory.trajectory_elements[1].angular_speed, 3, 0.0001);
+    EXPECT_NEAR(trajectory.trajectory_elements[2].angular_speed, sqrt(6), 0.0001);
+    EXPECT_NEAR(trajectory.trajectory_elements[3].angular_speed, 0, 0.0001);
+}
+
+TEST_F(TrajectoryPlannerTest, test_generate_maximum_speed_curve_straight_line)
```

Comment:
This test name is a bit confusing to me. Could you explain to me what you're testing here and I can try to think up on that's more intuitively obvious?

Reply:
This tests the function for 'generating the maximum speed curve of a straight line'

I've added a few more words to clarify